src/java.desktop/unix/native/libsplashscreen/splashscreen_sys.c

```diff
@@ -740,11 +738,10 @@ void
 SplashCreateThread(Splash * splash) {
     pthread_t thr;
     pthread_attr_t attr;
-    int rc;
 
     int rslt = pthread_attr_init(&attr);
     if (rslt != 0) return;
-    rc = pthread_create(&thr, &attr, SplashScreenThread, (void *) splash);
+    pthread_create(&thr, &attr, SplashScreenThread, (void *) splash);
```

Comment:
You don't think it would be better to check the return code?

Reply:
This code is devoid of pretty much all error handling and logging, but I agree that a simple fprintf on error would be useful.
Also doesn't a call like this trigger the warning about ignoring a function result, or have we disabled that one?

Reply:
> Also doesn't a call like this trigger the warning about ignoring a function result, or have we disabled that one?

Such a warning would be horrible to have enable. Then you would have to read the value of every function that thought it should return a value, even if it was just intended as some additional information that perhaps not everyone wanted to use. (And then you could not just store it into a variable and ignore it, since that would trigger the "unused variable" warning! ;-))

I know IntelliJ can provide warnings when ignoring the return value of a few select JDK methods that are often used incorrectly, but then they had to annotate these specially. I don't know if there is any similar kind of annotation for clang/gcc that can warn about ignoring return values that really, really shouldn't be ignored, but I'd guess not.

Reply:
Not entirely sure about clang, but casting to void to silence warnings should still work for gcc

Reply:
> You don't think it would be better to check the return code?

Hi , maybe we could check it ;  in HS code we mostly do  (but there we have UL).
But would be a separate PR I guess (splashscreen_sys.m on macOS has the same issue) .